### PR TITLE
Add DoNotUseIsOneAssertions rule

### PR DESCRIPTION
**Pull request by nakulpathak3**
Moving it over here as per https://github.com/Faire/backend/pull/144173#discussion_r1586541528. 

JIRA Ticket: https://fairewholesale.atlassian.net/browse/FD-208736

### CODE REVIEW

**Comment by nakulpathak3** [comment]
PR to remove the inverse rule from backend - https://github.com/Faire/backend/pull/146826

File: src/test/kotlin/com/faire/detekt/rules/DoNotUseIsOneAssertionsTest.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+private const val ISSUE_DESCRIPTION = "Do not use isOne(), use isEqualTo(1) instead."
+
+internal class DoNotUseIsOneAssertionsTest : AutoCorrectRuleTest<DoNotUseIsOneAssertions>(
+    { DoNotUseIsOneAssertions(it) },
+) {
+    @Test
+    fun `using isOne causes failure`() {
```
Comment (travisMiehm) [inline src/test/kotlin/com/faire/detekt/rules/DoNotUseIsOneAssertionsTest.kt:14]: Add a test for calling `isOne()` with a receiver context that is not an assertion method?

File: src/main/kotlin/com/faire/detekt/FaireRulesProvider.kt [hunk 2]
```diff
@@ -42,6 +43,7 @@ internal class FaireRulesProvider : RuleSetProvider {
       DoNotUseHasSizeForEmptyListInAssert(config),
       DoNotUseIsEqualToWhenArgumentIsOne(config),
       DoNotUseIsEqualToWhenArgumentIsZero(config),
+      DoNotUseIsOneAssertions(config),
```
[the remark this answers is not in the document]
Comment (nakulpathak3) [reply src/main/kotlin/com/faire/detekt/FaireRulesProvider.kt:46]: Is the way to have it disabled by default to just not add it to this file? I'm dropping that one for backend [in this PR](https://github.com/Faire/backend/pull/146826).

[the remark this answers is not in the document]
Comment (travisMiehm) [reply src/main/kotlin/com/faire/detekt/FaireRulesProvider.kt:46]: Talked offline, we'll let users of these rules just update their config file when they update the version.

**Review by travisMiehm — APPROVED** [review]

**Pull request merged**